Start menu

kgarc044 opened the pull request:
Hey randy, here is the pull request for the changes I made. That includes the start menu and the victory/gameover screen.

kgarc044 committed:
Start Menu

Added a main menu to the game with a Play Button (switches scenes to the game), Options button (which leads to options menu) and quit button (which quits the application.  There is also a back button which brings you back to the main menu if options is selected.

kgarc044 committed:
Victory Screen Setup

Started making text and button UIs for the victory screen.

kgarc044 committed:
Added Gameover Screen

Both Victory and Gameoever screens have buttons to restart the game or return to the main menu.

randymester commented on Turn Based RPG Tutorial/Assets/Scripts/FighterStats.cs:
```diff
@@ -119,4 +125,21 @@ public int CompareTo(object otherStats)
+
+    public void DisplayEndScreen(string unitName)
+    {
+
+        Debug.Log("This function calls");
+
```
Get rid of extra spacing and debugging

randymester commented on Turn Based RPG Tutorial/Assets/Scripts/FighterStats.cs:
```diff
@@ -119,4 +125,21 @@ public int CompareTo(object otherStats)
+        if (unitName == "Wizard Hero")
+        {
+            GameOverScreen.SetActive(true);
+            Debug.Log("GameOver");
```
Get rid of debugging

randymester commented on Turn Based RPG Tutorial/Assets/Scripts/FighterStats.cs:
```diff
@@ -119,4 +125,21 @@ public int CompareTo(object otherStats)
+        else
+        {
+            VictoryScreen.SetActive(true);
+            Debug.Log("VICTORY");
```
Get rid of debugging

randymester commented on Turn Based RPG Tutorial/Assets/Scripts/GameController.cs:
```diff
@@ -59,4 +61,6 @@ public void NextTurn()
             NextTurn();
         }
     }
+
+    
```
Get rid of extra spacing

randymester commented on Turn Based RPG Tutorial/Assets/Scripts/MainMenu.cs:
```diff
@@ -0,0 +1,30 @@
+
+    public void QuitGame()
+    {
+        Debug.Log("Quit");
```
Get rid of debugging

randymester reviewed (changes requested):
There are two additional requests I have that are not in the code.

1. When you click play, the giant's attack effects the player on the very first frame so it looks like the player starts with less health. Maybe you can add a brief wait period before the first turn so it's less jarring. Also maybe randomize who goes first? Not necessary, just a little extra.
2. I don't think the speed option actually does anything.

kgarc044 committed:
Added Delay before game start

The game waits a few seconds before the giant or the player can do any action.

kgarc044 committed:
Implemented Additional Requests

This should cover all the requests you mentioned. An additional script was added to fix the options menu. A coroutine was added to add delay for the game starts.

kgarc044 committed:
Final Commit

Removed the run button since it has no real functionality.
